includes/class-wcsatt-schemes.php

```diff
@@ -1,317 +1,166 @@
 <?php
 /**
- * Loading and filtering of subscription scheme settings.
+ * WCS_ATT_Schemes class
  *
- * @class  WCS_ATT_Schemes
- * @since  1.0.0
+ * @author   SomewhereWarm <info@somewherewarm.gr>
+ * @package  WooCommerce Subscribe All The Things
+ * @since    1.0.0
  */
 
-class WCS_ATT_Schemes {
-
-	/**
-	 * Returns the active cart-level subscription scheme id, or '0' if none is set.
-	 *
-	 * @return string
-	 */
-	public static function get_active_cart_subscription_scheme_id() {
-
-		return WC()->session->get( 'wcsatt-active-scheme-id', '0' );
-	}
-
-	/**
-	 * Returns the active subscription scheme of a cart item, or false if the cart item is a one-off purchase.
-	 *
-	 * @return string
-	 */
-	public static function get_active_subscription_scheme_id( $cart_item ) {
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
 
-		$active_scheme_id = isset( $cart_item[ 'wccsub_data' ][ 'active_subscription_scheme_id' ] ) ? $cart_item[ 'wccsub_data' ][ 'active_subscription_scheme_id' ] : '0';
+/**
+ * Loading and filtering of subscription scheme settings.
+ *
+ * @class    WCS_ATT_Schemes
+ * @version  2.0.0
+ */
+class WCS_ATT_Schemes {
 
-		return $active_scheme_id;
-	}
+	/*
+	|--------------------------------------------------------------------------
+	| Helpers (legacy)
+	|--------------------------------------------------------------------------
+	*/
 
 	/**
-	 * Returns the active subscription scheme of a cart item, or false if the cart item is a one-off purchase.
+	 * Returns a subscription scheme by id. Legacy.
 	 *
 	 * @return array
 	 */
-	public static function get_active_subscription_scheme( $cart_item ) {
+	public static function get_subscription_scheme_by_id( $id, $schemes ) {
```

Comment:
I think this function is a leftover from v1 and not used in v2 at all. In v1, "keys" were called "ids". This was changed in v2 to reflect the notion that "id" usually refers to a DB identifier.

Note that some filter names still use the `_id` suffix for compat with v1. I didn't touch those to avoid deprecating them.